compiler/rustc_middle/src/ty/context.rs

```diff
@@ -2440,7 +2443,7 @@ impl<'tcx> TyCtxt<'tcx> {
     /// Given the def_id of a Trait `trait_def_id` and the name of an associated item `assoc_name`
     /// returns true if the `trait_def_id` defines an associated item of name `assoc_name`.
     pub fn trait_may_define_assoc_item(self, trait_def_id: DefId, assoc_name: Ident) -> bool {
-        self.super_traits_of(trait_def_id).any(|trait_did| {
+        self.supertrait_def_ids(trait_def_id).any(|trait_did| {
```

Comment:
side-note: I kinda dislike that this is a method on `TyCtxt`. It makes it look like a query, and it's not like the other elaborator-like which are functions in `util::*`. :/

Reply:
Also I renamed this to `supertrait_def_ids` since `super_traits_of` makes it seem like it returns `PolyTraitRef`s which it explicit does not.